pkg-config.cmake

```diff
@@ -220,17 +220,30 @@ MACRO(_SETUP_PROJECT_PKG_CONFIG_FINALIZE)
   ENDIF()
 ENDMACRO(_SETUP_PROJECT_PKG_CONFIG_FINALIZE)
 
-# _PARSE_PKG_CONFIG_STRING (PKG_CONFIG_STRING _PKG_LIB_NAME_VAR _PKG_PREFIX_VAR)
+# _PARSE_PKG_CONFIG_STRING (PKG_CONFIG_STRING _PKG_LIB_NAME_VAR _PKG_PREFIX_VAR _PKG_CONFIG_STRING_NOSPACE_VAR)
 # ----------------------------------------------------------
 # 
-# Retrieve the library name and the prefix used for CMake variable names
-# from the pkg-config string. For instance, `my-package > 0.4` results in
+# Retrieve from the pkg-config string:
+# - the library name,
+# - the prefix used for CMake variable names,
+# - a variant without spaces around the operator (if there is an operator),
+#   as expected by cmake's CHECK_PKG_MODULE.
+# . For instance, `my-package > 0.4` results in
 # - _PKG_LIB_NAME_VAR <- my-package
 # - _PKG_PREFIX_VAR <- MY_PACKAGE
-MACRO(_PARSE_PKG_CONFIG_STRING PKG_CONFIG_STRING _PKG_LIB_NAME_VAR _PKG_PREFIX_VAR)
-  # Retrieve the left part of the equation to get package name.
-  STRING(REGEX MATCH "[^<>= ]+" ${_PKG_LIB_NAME_VAR} "${PKG_CONFIG_STRING}")
-  # And transform it into a valid variable prefix.
+# - _PKG_CONFIG_STRING_NOSPACE_VAR <- `my-package>0.4`
+# `my-package` results in
+# - _PKG_LIB_NAME_VAR <- my-package
+# - _PKG_PREFIX_VAR <- MY_PACKAGE
+# - _PKG_CONFIG_STRING_NOSPACE_VAR <- `my-package`
+MACRO(_PARSE_PKG_CONFIG_STRING PKG_CONFIG_STRING _PKG_LIB_NAME_VAR _PKG_PREFIX_VAR _PKG_CONFIG_NOSPACE_VAR)
+  # Decompose the equation
+  STRING(REGEX MATCH "([^ ]+)( (>|>=|=|<=|<) (.*))?" _FOO "${PKG_CONFIG_STRING}")
+  # Reconstruct the equation, without the space around the operator
+  SET(${_PKG_CONFIG_NOSPACE_VAR} "${CMAKE_MATCH_1}${CMAKE_MATCH_3}${CMAKE_MATCH_4}")
```

Comment:
I think the access to regex groups was added in a recent version of CMake. CMake 3.2 does not support it (at least it is not in the [doc](https://cmake.org/cmake/help/v3.2/command/string.html)).
You can work around it with `STRING(REGEX REPLACE ...)` which support regex groups.

Reply:
The 3.2 version doc says

```
 ()        Saves a matched subexpression, which can be referenced
           in the REGEX REPLACE operation. Additionally it is saved
           by all regular expression-related commands, including
           e.g. if( MATCHES ), in the variables CMAKE_MATCH_(0..9).
```

I'll try to find a way to get can old cmake version on the CI.

Reply:
You are right. I missed it. Sorry for the noise.

Reply:
> I'll try to find a way to get an old cmake version on the CI.

- ubuntu 16.04 ships with cmake 3.5.
- [github actions' ubuntu 16.04](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/software-installed-on-github-hosted-runners#ubuntu-1604-lts) ships with cmake 3.12.
- [The pip package](https://pypi.org/project/cmake/) seems to start at [3.6](https://pypi.org/project/cmake/0.1.0/)


If old cmake versions need to be tested, the simplest solution might be yo use the kitware binaries (eg. https://cmake.org/files/v3.2/)

Anyway, there is no hurry as long as the project has no CI in place.